### PR TITLE
types/hubot: added catchAll

### PR DESCRIPTION
**Pull request by Eeemil**
catchAll is a method for replying to messages which does not match any messages matched by hear or respond

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [-] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/hubotio/hubot/blob/master/src/robot.js#L223>>
- [-] Increase the version number in the header if appropriate.
- [-] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by dirk — APPROVED** [review]
Looks good to me! Thanks for the enhancement @Eeemil. 😄

File: types/hubot/index.d.ts [hunk 1]
```diff
@@ -40,6 +41,8 @@ declare namespace Hubot {
         readonly adapter: A;
 
         constructor(adapterPath: string, adapter: string, httpd: boolean, name: string, alias?: string);
+        catchAll(callback: ListenerCallback<this>): void;
+        catchAll(options: any, callback: ListenerCallback<this>): void;
```
Comment (jessetrinity) [inline types/hubot/index.d.ts]: Can options be optional here instead?

Comment (Eeemil) [reply types/hubot/index.d.ts]: Yes, and it can be optional for the hear and response methods as well, should I change them all? :)

Comment (jessetrinity) [reply types/hubot/index.d.ts]: It would be really cool if you did :)

Comment (Eeemil) [reply types/hubot/index.d.ts]: I realize now that I might have been wrong. I'm quite new to TypeScript, the options parameter is always optional in the JS code, but for TypeScript, optional parameters needs to come after mandatory parameters. I don't know how to make the options parameter optional, if possible...

Comment (jessetrinity) [reply types/hubot/index.d.ts]: Putting `options?:` would make it optional, though optional parameters do need to come after all non-optional parameters. [Optional parameters vs overloads here](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-optional-parameters). There may have been some reason you did not want to do this here? Probably not a big deal either way.

**Pull request merged**